### PR TITLE
Add gamma distribution

### PR DESCRIPTION
**Pull request by YoshikawaMasashi**
split #1177

### CODE REVIEW

**Comment by YoshikawaMasashi** [comment]
jenkins, test this please.

**Comment by YoshikawaMasashi** [comment]
jenkins, test this please.

File: cupy/random/__init__.py [hunk 1]
```diff
@@ -18,6 +18,7 @@ def bytes(length):
 # import class and function
 from cupy.random.distributions import beta  # NOQA
 from cupy.random.distributions import binomial  # NOQA
+from cupy.random.distributions import gamma  # NOQA
```
Comment (toslunar) [inline cupy/random/__init__.py]: Please fix the import order.

File: cupy/random/_kernels.py [hunk 2]
```diff
@@ -299,6 +299,53 @@
...
+            } while (V <= 0.0);
+            V = V*V*V;
+            U = rk_double(state);
+            if (U < 1.0 - 0.0331*(X*X)*(X*X)) return (b*V);
+            if (log(U) < 0.5*X*X + b*(1. - V + log(V))) return (b*V);
+        }
+    }
+}
+'''
+
```
Comment (toslunar) [inline cupy/random/_kernels.py]: It seems these kernels are already in the file.

File: cupy/random/distributions.py [hunk 3]
```diff
@@ -81,7 +109,7 @@ def dirichlet(alpha, size=None, dtype=float):
             :class:`numpy.float64` types are allowed.
 
     Returns:
-        cupy.ndarray: Samples drawn from the dirichret destribution.
+        cupy.ndarray: Samples drawn from the gamma destribution.
```
Comment (toslunar) [inline cupy/random/distributions.py]: Please revert the change to this line.

File: cupy/random/distributions.py [hunk 4]
```diff
@@ -61,6 +61,34 @@ def binomial(n, p, size=None, dtype=int):
...
+
+    Returns an array of samples drawn from the gamma distribution. Its
+    probability density function is defined as
+
+    .. math::
+       f(x) = \\frac{1}{\\Gamma(k)\\theta^k}x^{k-1}e^{-x/\\theta},
+
+    Args:
+        shape (float): Parameter of the gamma distribution :math:`k`.
+        scale (float): Parameter of the gamma distribution :math:`\\theta`
```
Comment (toslunar) [inline cupy/random/distributions.py]: It seems `shape` and `scale` can be `cupy.ndarray`s.

File: tests/cupy_tests/random_tests/test_distributions.py [hunk 5]
```diff
@@ -79,6 +79,30 @@ def test_dirichlet(self, alpha_dtype, dtype):
...
+
+    def check_distribution(self, dist_func, shape_dtype, scale_dtype, dtype):
+        shape = cupy.ones(self.shape_shape, dtype=shape_dtype)
+        scale = cupy.ones(self.scale_shape, dtype=scale_dtype)
+        out = dist_func(shape, scale, self.shape, dtype)
+        self.assertEqual(self.shape, out.shape)
+        self.assertEqual(out.dtype, dtype)
+
+    @cupy.testing.for_float_dtypes('shape_dtype')
+    @cupy.testing.for_float_dtypes('scale_dtype')
```
Comment (toslunar) [inline tests/cupy_tests/random_tests/test_distributions.py]: Why don't use `@testing.for_dtypes_combination`?

File: docs/source/reference/random.rst [hunk 6]
```diff
@@ -38,6 +38,7 @@ Distributions
 
    cupy.random.beta
    cupy.random.binomial
+   cupy.random.gamma
```
Comment (toslunar) [inline docs/source/reference/random.rst]: Please fix the order.

**Comment by YoshikawaMasashi** [comment]
jenkins, test this please.
